src/L1/SwapAndBridge.sol

```diff
@@ -0,0 +1,108 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity 0.8.23;
+
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+
+/// @title IL1StandardBridge - L1 Standard Bridge interface
+/// @notice This contract is used to transfer L1 tokens to the L2 network as L2 tokens.
+interface IL1StandardBridge {
+    /// Deposits L1 Lisk tokens into a target account on L2 network.
+    /// @param _l1Token L1 token address.
+    /// @param _l2Token L2 token address.
+    /// @param _to Target account address on L2 network.
+    /// @param _amount Amount of L1 tokens to be transferred.
+    /// @param _minGasLimit Minimum gas limit for the deposit message on L2.
+    /// @param _extraData Optional data to forward to L2. Data supplied here will not be used to
+    ///                   execute any code on L2 and is only emitted as extra data for the
+    ///                   convenience of off-chain tooling.
+    function depositERC20To(
+        address _l1Token,
+        address _l2Token,
+        address _to,
+        uint256 _amount,
+        uint32 _minGasLimit,
+        bytes calldata _extraData
+    )
+        external;
+}
+
+/// @title SwapAndBridge
+/// @notice SwapAndBridge is the utility contract that allows to swap ETH to a wrapped LST and bridge it to L2.
+///         It is designed to be used as a part of the Lisk L2 ecosystem.
+contract SwapAndBridge {
+    /// @notice Minimum amount of gas to be used for the deposit message on L2.
+    uint32 public constant MIN_DEPOSIT_GAS = 0;
```

Comment:
Once again, why can be `MIN_DEPOSIT_GAS` set to zero?

Reply:
It will be overridden in the bridge to a minimum calculated from the message. Having a non-zero min gas can have the effect of having the user pay an extra, which I think we should not set here.

Reply:
Should we add similar comment inside file for this variable?
I think this is not something reader will know in advance without having a deep knowledge about how bridging works.

Reply:
this is the relevant link btw:https://github.com/ethereum-optimism/optimism/blob/develop/packages/contracts-bedrock/src/universal/CrossDomainMessenger.sol#L187

I will add a comment, but there is this detail that in principle here we only require the bridge to follow the `depositERC20To` interface, with no knowledge of the bridge internals. Then of course, the tandard bridge is an important example and we could add the comment anyway